src/sync/server.rs

```diff
@@ -137,20 +138,22 @@ fn start_method_handler_thread(
             let mh;
             let buf;
             match result {
-                Ok((x, y)) => {
+                Ok((x, Ok(y))) => {
                     mh = x;
                     buf = y;
                 }
+                Ok((mh, Err(e))) => {
+                    if let Err(x) = respond_error_to_channel(mh.stream_id, e, res_tx.clone()) {
+                        debug!("response_error_to_channel get error {:?}", x);
+                        quit_connection(quit, control_tx);
```

Comment:
Why we should quit the connection only when get wrong data, such as data length was out of the ttrpc max size?

Reply:
Not because the length exceeded but because sending the error message failed